aptos-move/move-examples/defi/aptoswap/init/sources/pool_manager.move

```diff
@@ -0,0 +1,43 @@
+module aptoswap::init {
+    use std::signer;
+    use aptos_framework::object::{Self, ExtendRef};
+
+    const ERR_NOT_APTOSWAP_ADMIN: u64 = 1;
+    const ERR_EXTEND_REF_NOT_MATCH: u64 = 2;
+
+    struct PoolManagerExtendRef has key {
+        ref: ExtendRef
+    }
+
+    /// Creates new object for Aptoswap to be the pool manager object where the lp code will be deployed at.
+    /// Can only be executed only from Aptoswap admin account.
+    public entry fun initialize_lp_account(
```

Comment:
This is a strange flow. Why create a different lp object every time, store ExtendRef in admin account, and then retrieve + take over ExtendRef later?

Why not allow liquidity_pool to call this directly to create a liquidity pool object, then still do the management here (store extend ref, provide a signer for extension that's a friend only function and can be called by modules in the same package)?